### ISSUE TITLE
[🐛 Bug]: new selenium doesn't find Firefox

### ISSUE THREAD

**Issue by noelo-cohelo** [SeleniumHQ/selenium#13366]
### What happened?

After Selenium upgrade from 3.13.0, GeckoDriver stopped finding Firefox. It expects it to be installed as snap, but I just have one from yum repository. Uncommenting the commented line from code sample is a workaround for this problem. This seems like a regression.

Following issues might be caused by the same:
https://github.com/SeleniumHQ/selenium/issues/12495
https://github.com/SeleniumHQ/selenium/issues/11414

There is also a StackExchange topic: https://sqa.stackexchange.com/questions/51464/selenium-latest-version-is-not-able-to-launch-firefox-browser-in-ubuntu-20-04

### How can we reproduce the issue?

```shell
FirefoxOptions options = new FirefoxOptions();
// options.setBinary("/usr/bin/firefox");
FirefoxDriver driver = new FirefoxDriver(options);
driver.get("https://www.google.com");
```

### Relevant log output

```shell
/opt/c2p/jdk1.8.0/bin/java -javaagent:/var/lib/snapd/snap/intellij-idea-community/467/lib/idea_rt.jar=42082:/var/lib/snapd/snap/intellij-idea-community/467/bin -Dfile.encoding=UTF-8 -classpath /opt/c2p/jdk1.8.0/jre/lib/charsets.jar:/opt/c2p/jdk1.8.0/jre/lib/deploy.jar:/opt/c2p/jdk1.8.0/jre/lib/ext/cldrdata.jar:/opt/c2p/jdk1.8.0/jre/lib/ext/dnsns.jar:/opt/c2p/jdk1.8.0/jre/lib/ext/jaccess.jar:/opt/c2p/jdk1.8.0/jre/lib/ext/jfxrt.jar:/opt/c2p/jdk1.8.0/jre/lib/ext/localedata.jar:/opt/c2p/jdk1.8.0/jre/lib/ext/nashorn.jar:/opt/c2p/jdk1.8.0/jre/lib/ext/sunec.jar:/opt/c2p/jdk1.8.0/jre/lib/ext/sunjce_provider.jar:/opt/c2p/jdk1.8.0/jre/lib/ext/sunpkcs11.jar:/opt/c2p/jdk1.8.0/jre/lib/ext/zipfs.jar:/opt/c2p/jdk1.8.0/jre/lib/javaws.jar:/opt/c2p/jdk1.8.0/jre/lib/jce.jar:/opt/c2p/jdk1.8.0/jre/lib/jfr.jar:/opt/c2p/jdk1.8.0/jre/lib/jfxswt.jar:/opt/c2p/jdk1.8.0/jre/lib/jsse.jar:/opt/c2p/jdk1.8.0/jre/lib/management-agent.jar:/opt/c2p/jdk1.8.0/jre/lib/plugin.jar:/opt/c2p/jdk1.8.0/jre/lib/resources.jar:/opt/c2p/jdk1.8.0/jre/lib/rt.jar:/home/iid/git/csgpp/tests/GPPMarketSmokeTests/target/classes:/home/iid/git/csgpp/patch/opf/OPFResources/target/classes:/home/iid/git/csgpp/patch/sepa/SEPAResources/target/classes:/home/iid/git/csgpp/patch/sepa/SEPASOResources/target/classes:/home/iid/git/csgpp/patch/sic/SICResources/target/classes:/home/iid/git/csgpp/patch/xct/XCTResources/target/classes:/home/iid/git/csgpp/patch/opf/OPFPaymentDataAccess/target/classes:/home/iid/git/csgpp/patch/opf/OPFCommonEntityModel/target/classes:/home/iid/git/csgpp/patch/opf/OPFBaseEntityModel/target/classes:/home/iid/git/csgpp/patch/opf/OPFPaymentEntityModel/target/classes:/home/iid/.m2/repository/com/clear2pay/opf/OPFPaymentPlatform/11.9.1.2-436599/OPFPaymentPlatform-11.9.1.2-436599.jar:/home/iid/git/csgpp/patch/opf/OPFCommonDataAccess/target/classes:/home/iid/.m2/repository/net/sf/ehcache/ehcache/2.9.1/ehcache-2.9.1.jar:/home/iid/.m2/repository/org/slf4j/slf4j-api/1.7.14/slf4j-api-1.7.14.jar:/home/iid/git/csgpp/patch/opf/OPFBasePlatform/target/classes:/home/iid/.m2/repository/org/apache/commons/commons-lang3/3.2.1/commons-lang3-3.2.1.jar:/home/iid/.m2/repository/commons-jxpath/commons-jxpath/1.3/commons-jxpath-1.3.jar:/home/iid/.m2/repository/commons-configuration/commons-configuration/1.10/commons-configuration-1.10.jar:/home/iid/.m2/repository/commons-lang/commons-lang/2.6/commons-lang-2.6.jar:/home/iid/.m2/repository/commons-collections/commons-collections/3.2.2/commons-collections-3.2.2.jar:/home/iid/.m2/repository/commons-codec/commons-codec/1.15/commons-codec-1.15.jar:/home/iid/.m2/repository/com/clear2pay/core/whitebox/0.2-21/whitebox-0.2-21.jar:/home/iid/.m2/repository/com/codahale/metrics/metrics-core/3.0.2/metrics-core-3.0.2.jar:/home/iid/.m2/repository/com/clear2pay/core/whitebox-healthcheck/0.1-8/whitebox-healthcheck-0.1-8.jar:/home/iid/.m2/repository/org/antlr/antlr4-runtime/4.5/antlr4-runtime-4.5.jar:/home/iid/.m2/repository/org/abego/treelayout/org.abego.treelayout.core/1.0.1/org.abego.treelayout.core-1.0.1.jar:/home/iid/.m2/repository/org/json/json/20231013/json-20231013.jar:/home/iid/git/csgpp/patch/opf/OPFBaseDataAccess/target/classes:/home/iid/.m2/repository/org/hibernate/hibernate-core/4.3.11.Final/hibernate-core-4.3.11.Final.jar:/home/iid/.m2/repository/org/jboss/logging/jboss-logging/3.1.3.GA/jboss-logging-3.1.3.GA.jar:/home/iid/.m2/repository/org/jboss/logging/jboss-logging-annotations/1.2.0.Beta1/jboss-logging-annotations-1.2.0.Beta1.jar:/home/iid/.m2/repository/org/hibernate/common/hibernate-commons-annotations/4.0.5.Final/hibernate-commons-annotations-4.0.5.Final.jar:/home/iid/.m2/repository/org/hibernate/javax/persistence/hibernate-jpa-2.1-api/1.0.0.Final/hibernate-jpa-2.1-api-1.0.0.Final.jar:/home/iid/.m2/repository/antlr/antlr/2.7.7/antlr-2.7.7.jar:/home/iid/.m2/repository/org/jboss/jandex/1.1.0.Final/jandex-1.1.0.Final.jar:/home/iid/.m2/repository/org/dom4j/dom4j/2.1.3/dom4j-2.1.3.jar:/home/iid/.m2/repository/com/google/guava/guava/32.0.1-jre/guava-32.0.1-jre.jar:/home/iid/.m2/repository/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar:/home/iid/.m2/repository/com/google/guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar:/home/iid/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar:/home/iid/.m2/repository/org/checkerframework/checker-qual/3.33.0/checker-qual-3.33.0.jar:/home/iid/.m2/repository/com/google/errorprone/error_prone_annotations/2.18.0/error_prone_annotations-2.18.0.jar:/home/iid/.m2/repository/com/google/j2objc/j2objc-annotations/2.8/j2objc-annotations-2.8.jar:/home/iid/.m2/repository/com/clear2pay/opf/OPFPortStandalone/11.9.1.2-436599/OPFPortStandalone-11.9.1.2-436599.jar:/home/iid/git/csgpp/patch/opf/OPFPort/target/classes:/home/iid/.m2/repository/commons-logging/commons-logging-api/1.1/commons-logging-api-1.1.jar:/home/iid/.m2/repository/cglib/cglib-nodep/3.1/cglib-nodep-3.1.jar:/home/iid/.m2/repository/xalan/xalan/2.7.2/xalan-2.7.2.jar:/home/iid/.m2/repository/xalan/serializer/2.7.2/serializer-2.7.2.jar:/home/iid/git/csgpp/sources/GPPMarketEntityModel/target/classes:/home/iid/git/csgpp/sources/GPPCommonEntityModel/target/classes:/home/iid/git/csgpp/patch/sepa/SEPAEntityModel/target/classes:/home/iid/.m2/repository/com/clear2pay/sepa/SEPAPlatform/9.1.2.0-451877/SEPAPlatform-9.1.2.0-451877.jar:/home/iid/.m2/repository/commons-beanutils/commons-beanutils/1.8.3/commons-beanutils-1.8.3.jar:/home/iid/git/csgpp/patch/sic/SICEntityModel/target/classes:/home/iid/git/csgpp/patch/xct/XCTEntityModel/target/classes:/home/iid/.m2/repository/com/clear2pay/opf/StandingOrderEntityModel/11.9.1.2-436599/StandingOrderEntityModel-11.9.1.2-436599.jar:/home/iid/git/csgpp/patch/xct/XCTPlatform/target/classes:/home/iid/git/csgpp/sources/GPPIdentificationCommons/target/classes:/home/iid/.m2/repository/com/clear2pay/opf/OPFIdentificationMiddleware/11.9.1.2-436599/OPFIdentificationMiddleware-11.9.1.2-436599.jar:/home/iid/git/csgpp/patch/opf/OPFIdentification/target/classes:/home/iid/.m2/repository/com/clear2pay/opf/OPFRealm/11.9.1.2-436599/OPFRealm-11.9.1.2-436599.jar:/home/iid/.m2/repository/commons-io/commons-io/2.8.0/commons-io-2.8.0.jar:/home/iid/.m2/repository/commons-pool/commons-pool/1.6/commons-pool-1.6.jar:/home/iid/git/csgpp/tests/GoldFishApi/target/classes:/home/iid/.m2/repository/com/oracle/weblogic/ojdbc7/12.1.3-0-0/ojdbc7-12.1.3-0-0.jar:/home/iid/git/csgpp/tests/GoldFishCommon/target/classes:/home/iid/.m2/repository/log4j/log4j/1.2.17/log4j-1.2.17.jar:/home/iid/.m2/repository/org/glassfish/jersey/core/jersey-common/2.26/jersey-common-2.26.jar:/home/iid/.m2/repository/org/glassfish/hk2/osgi-resource-locator/1.0.1/osgi-resource-locator-1.0.1.jar:/home/iid/git/csgpp/sources/Resources/target/classes:/home/iid/.m2/repository/com/clear2pay/opf/OPFUnitTests/11.9.1.2-436599/OPFUnitTests-11.9.1.2-436599.jar:/home/iid/git/csgpp/patch/opf/OPFCommonServices/target/classes:/home/iid/.m2/repository/com/clear2pay/opf/OPFCommonServicesClient/11.9.1.2-436599/OPFCommonServicesClient-11.9.1.2-436599.jar:/home/iid/.m2/repository/com/clear2pay/opf/tools/bpmn/flow-task-annotation/1.1.9-944766/flow-task-annotation-1.1.9-944766.jar:/home/iid/.m2/repository/com/clear2pay/osgi/workflow/osgi-workflow-api-bundle/1.3.2-21-f0ada5c/osgi-workflow-api-bundle-1.3.2-21-f0ada5c.jar:/home/iid/.m2/repository/com/clear2pay/osgi/workflow/osgi-workflow-service-api/1.3.2-21-f0ada5c/osgi-workflow-service-api-1.3.2-21-f0ada5c.jar:/home/iid/.m2/repository/com/clear2pay/opf/OPFCommonOrchestrationClient/11.9.1.2-436599/OPFCommonOrchestrationClient-11.9.1.2-436599.jar:/home/iid/.m2/repository/joda-time/joda-time/2.9.1/joda-time-2.9.1.jar:/home/iid/.m2/repository/com/fasterxml/woodstox/woodstox-core/6.4.0/woodstox-core-6.4.0.jar:/home/iid/.m2/repository/org/apache/velocity/velocity-engine-core/2.3/velocity-engine-core-2.3.jar:/home/iid/.m2/repository/com/clear2pay/opf/OPFCommonServicesTestTools/11.9.1.2-436599/OPFCommonServicesTestTools-11.9.1.2-436599.jar:/home/iid/.m2/repository/com/clear2pay/opf/OPFBaseDataAccessTestTools/11.9.1.2-436599/OPFBaseDataAccessTestTools-11.9.1.2-436599.jar:/home/iid/.m2/repository/com/clear2pay/opf/OPFBasePlatformTestTools/11.9.1.2-436599/OPFBasePlatformTestTools-11.9.1.2-436599.jar:/home/iid/.m2/repository/junit/junit/4.12/junit-4.12.jar:/home/iid/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar:/home/iid/git/csgpp/tests/GPPTestVisibilityCommon/target/classes:/home/iid/git/csgpp/sources/GPPServices/target/classes:/home/iid/git/csgpp/sources/GPPServicesClient/target/classes:/home/iid/git/csgpp/patch/sepa/SEPABusinessServicesClient/target/classes:/home/iid/git/csgpp/patch/sepa/SEPADataAccess/target/classes:/home/iid/git/csgpp/sources/GPPCommonDataAccess/target/classes:/home/iid/.m2/repository/io/prometheus/simpleclient_common/0.0.15/simpleclient_common-0.0.15.jar:/home/iid/.m2/repository/io/prometheus/simpleclient/0.0.15/simpleclient-0.0.15.jar:/home/iid/.m2/repository/com/csg/pay/gpp/pil/code/bom/pay_gpp_lsc_inbound_lib/1.0.4/pay_gpp_lsc_inbound_lib-1.0.4.jar:/home/iid/git/csgpp/sources/GPPIdentification/target/classes:/home/iid/git/csgpp/patch/opf/OPFPaymentServices/target/classes:/home/iid/git/csgpp/patch/opf/OPFPaymentServicesClient/target/classes:/home/iid/.m2/repository/org/apache/poi/poi-ooxml-full/5.2.3/poi-ooxml-full-5.2.3.jar:/home/iid/.m2/repository/org/apache/xmlbeans/xmlbeans/5.1.1/xmlbeans-5.1.1.jar:/home/iid/.m2/repository/com/csg/pay/gpp/pil/code/c2p/pay_gpp_inbound_lib/1.0.2/pay_gpp_inbound_lib-1.0.2.jar:/home/iid/.m2/repository/org/apache/poi/poi-ooxml/5.2.3/poi-ooxml-5.2.3.jar:/home/iid/.m2/repository/org/apache/poi/poi/5.2.3/poi-5.2.3.jar:/home/iid/.m2/repository/org/apache/commons/commons-math3/3.6.1/commons-math3-3.6.1.jar:/home/iid/.m2/repository/com/zaxxer/SparseBitSet/1.2/SparseBitSet-1.2.jar:/home/iid/.m2/repository/org/apache/commons/commons-compress/1.21/commons-compress-1.21.jar:/home/iid/.m2/repository/com/github/virtuald/curvesapi/1.07/curvesapi-1.07.jar:/home/iid/.m2/repository/org/apache/logging/log4j/log4j-api/2.18.0/log4j-api-2.18.0.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-java/4.9.1/selenium-java-4.9.1.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-api/4.9.1/selenium-api-4.9.1.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-chrome-driver/4.9.1/selenium-chrome-driver-4.9.1.jar:/home/iid/.m2/repository/com/google/auto/service/auto-service-annotations/1.0.1/auto-service-annotations-1.0.1.jar:/home/iid/.m2/repository/com/google/auto/service/auto-service/1.0.1/auto-service-1.0.1.jar:/home/iid/.m2/repository/com/google/auto/auto-common/1.2/auto-common-1.2.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-chromium-driver/4.9.1/selenium-chromium-driver-4.9.1.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-json/4.9.1/selenium-json-4.9.1.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-manager/4.9.1/selenium-manager-4.9.1.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-devtools-v111/4.9.1/selenium-devtools-v111-4.9.1.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-devtools-v112/4.9.1/selenium-devtools-v112-4.9.1.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-devtools-v113/4.9.1/selenium-devtools-v113-4.9.1.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-devtools-v85/4.9.1/selenium-devtools-v85-4.9.1.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-edge-driver/4.9.1/selenium-edge-driver-4.9.1.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-firefox-driver/4.9.1/selenium-firefox-driver-4.9.1.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-http/4.9.1/selenium-http-4.9.1.jar:/home/iid/.m2/repository/dev/failsafe/failsafe/3.3.1/failsafe-3.3.1.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-ie-driver/4.9.1/selenium-ie-driver-4.9.1.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-remote-driver/4.9.1/selenium-remote-driver-4.9.1.jar:/home/iid/.m2/repository/com/beust/jcommander/1.82/jcommander-1.82.jar:/home/iid/.m2/repository/io/netty/netty-buffer/4.1.91.Final/netty-buffer-4.1.91.Final.jar:/home/iid/.m2/repository/io/netty/netty-codec-http/4.1.91.Final/netty-codec-http-4.1.91.Final.jar:/home/iid/.m2/repository/io/netty/netty-codec/4.1.91.Final/netty-codec-4.1.91.Final.jar:/home/iid/.m2/repository/io/netty/netty-handler/4.1.91.Final/netty-handler-4.1.91.Final.jar:/home/iid/.m2/repository/io/netty/netty-common/4.1.91.Final/netty-common-4.1.91.Final.jar:/home/iid/.m2/repository/io/netty/netty-transport-classes-epoll/4.1.91.Final/netty-transport-classes-epoll-4.1.91.Final.jar:/home/iid/.m2/repository/io/netty/netty-transport-classes-kqueue/4.1.91.Final/netty-transport-classes-kqueue-4.1.91.Final.jar:/home/iid/.m2/repository/io/netty/netty-transport-native-epoll/4.1.91.Final/netty-transport-native-epoll-4.1.91.Final.jar:/home/iid/.m2/repository/io/netty/netty-transport-native-kqueue/4.1.91.Final/netty-transport-native-kqueue-4.1.91.Final.jar:/home/iid/.m2/repository/io/netty/netty-transport-native-unix-common/4.1.91.Final/netty-transport-native-unix-common-4.1.91.Final.jar:/home/iid/.m2/repository/io/netty/netty-transport/4.1.91.Final/netty-transport-4.1.91.Final.jar:/home/iid/.m2/repository/io/netty/netty-resolver/4.1.91.Final/netty-resolver-4.1.91.Final.jar:/home/iid/.m2/repository/io/opentelemetry/opentelemetry-api/1.25.0/opentelemetry-api-1.25.0.jar:/home/iid/.m2/repository/io/opentelemetry/opentelemetry-context/1.25.0/opentelemetry-context-1.25.0.jar:/home/iid/.m2/repository/io/opentelemetry/opentelemetry-exporter-logging/1.25.0/opentelemetry-exporter-logging-1.25.0.jar:/home/iid/.m2/repository/io/opentelemetry/opentelemetry-sdk-metrics/1.25.0/opentelemetry-sdk-metrics-1.25.0.jar:/home/iid/.m2/repository/io/opentelemetry/opentelemetry-sdk-logs/1.25.0-alpha/opentelemetry-sdk-logs-1.25.0-alpha.jar:/home/iid/.m2/repository/io/opentelemetry/opentelemetry-api-logs/1.25.0-alpha/opentelemetry-api-logs-1.25.0-alpha.jar:/home/iid/.m2/repository/io/opentelemetry/opentelemetry-api-events/1.25.0-alpha/opentelemetry-api-events-1.25.0-alpha.jar:/home/iid/.m2/repository/io/opentelemetry/opentelemetry-sdk-common/1.25.0/opentelemetry-sdk-common-1.25.0.jar:/home/iid/.m2/repository/io/opentelemetry/opentelemetry-sdk-extension-autoconfigure-spi/1.25.0/opentelemetry-sdk-extension-autoconfigure-spi-1.25.0.jar:/home/iid/.m2/repository/io/opentelemetry/opentelemetry-sdk-extension-autoconfigure/1.25.0-alpha/opentelemetry-sdk-extension-autoconfigure-1.25.0-alpha.jar:/home/iid/.m2/repository/io/opentelemetry/opentelemetry-sdk-trace/1.25.0/opentelemetry-sdk-trace-1.25.0.jar:/home/iid/.m2/repository/io/opentelemetry/opentelemetry-sdk/1.25.0/opentelemetry-sdk-1.25.0.jar:/home/iid/.m2/repository/io/opentelemetry/opentelemetry-semconv/1.25.0-alpha/opentelemetry-semconv-1.25.0-alpha.jar:/home/iid/.m2/repository/io/ous/jtoml/2.0.0/jtoml-2.0.0.jar:/home/iid/.m2/repository/net/bytebuddy/byte-buddy/1.14.4/byte-buddy-1.14.4.jar:/home/iid/.m2/repository/org/apache/commons/commons-exec/1.3/commons-exec-1.3.jar:/home/iid/.m2/repository/org/asynchttpclient/async-http-client/2.12.3/async-http-client-2.12.3.jar:/home/iid/.m2/repository/org/asynchttpclient/async-http-client-netty-utils/2.12.3/async-http-client-netty-utils-2.12.3.jar:/home/iid/.m2/repository/io/netty/netty-codec-socks/4.1.60.Final/netty-codec-socks-4.1.60.Final.jar:/home/iid/.m2/repository/io/netty/netty-handler-proxy/4.1.60.Final/netty-handler-proxy-4.1.60.Final.jar:/home/iid/.m2/repository/io/netty/netty-transport-native-epoll/4.1.60.Final/netty-transport-native-epoll-4.1.60.Final-linux-x86_64.jar:/home/iid/.m2/repository/io/netty/netty-transport-native-kqueue/4.1.60.Final/netty-transport-native-kqueue-4.1.60.Final-osx-x86_64.jar:/home/iid/.m2/repository/org/reactivestreams/reactive-streams/1.0.3/reactive-streams-1.0.3.jar:/home/iid/.m2/repository/com/typesafe/netty/netty-reactive-streams/2.0.4/netty-reactive-streams-2.0.4.jar:/home/iid/.m2/repository/com/sun/activation/jakarta.activation/1.2.2/jakarta.activation-1.2.2.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-safari-driver/4.9.1/selenium-safari-driver-4.9.1.jar:/home/iid/.m2/repository/org/seleniumhq/selenium/selenium-support/4.9.1/selenium-support-4.9.1.jar:/home/iid/.m2/repository/org/apache/httpcomponents/httpclient/4.5.9/httpclient-4.5.9.jar:/home/iid/.m2/repository/org/apache/httpcomponents/httpcore/4.4.11/httpcore-4.4.11.jar:/home/iid/.m2/repository/commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar:/home/iid/.m2/repository/org/reflections/reflections/0.9.10/reflections-0.9.10.jar:/home/iid/.m2/repository/org/javassist/javassist/3.19.0-GA/javassist-3.19.0-GA.jar:/home/iid/.m2/repository/com/google/code/findbugs/annotations/2.0.1/annotations-2.0.1.jar:/home/iid/.m2/repository/org/glassfish/jersey/core/jersey-server/2.26/jersey-server-2.26.jar:/home/iid/.m2/repository/org/glassfish/jersey/core/jersey-client/2.26/jersey-client-2.26.jar:/home/iid/.m2/repository/org/glassfish/jersey/media/jersey-media-jaxb/2.26/jersey-media-jaxb-2.26.jar:/home/iid/.m2/repository/javax/annotation/javax.annotation-api/1.2/javax.annotation-api-1.2.jar:/home/iid/.m2/repository/org/glassfish/hk2/external/javax.inject/2.5.0-b42/javax.inject-2.5.0-b42.jar:/home/iid/.m2/repository/javax/validation/validation-api/1.1.0.Final/validation-api-1.1.0.Final.jar:/home/iid/.m2/repository/org/glassfish/jersey/containers/jersey-container-servlet-core/2.26/jersey-container-servlet-core-2.26.jar:/home/iid/.m2/repository/org/glassfish/jersey/containers/jersey-container-servlet/2.26/jersey-container-servlet-2.26.jar:/home/iid/.m2/repository/com/csg/pay/gpp/pil/code/sbr/PB_GPP_PaymentOrderBackend_4/1.0.1/PB_GPP_PaymentOrderBackend_4-1.0.1.jar:/home/iid/.m2/repository/com/csg/pay/gpp/pil/code/sbr/PB_GPP_PaymentOrderBackend_3/19.0.0/PB_GPP_PaymentOrderBackend_3-19.0.0.jar:/home/iid/.m2/repository/com/csg/pay/gpp/pil/code/sbr/PB_GPP_PaymentOrderBackend_2/1.0.0/PB_GPP_PaymentOrderBackend_2-1.0.0.jar:/home/iid/git/csgpp/tests/GPPCommonTestUtils/target/classes:/home/iid/git/csgpp/sources/GPPCommonComponents/target/classes:/home/iid/.m2/repository/org/freemarker/freemarker/2.3.23/freemarker-2.3.23.jar:/home/iid/.m2/repository/org/awaitility/awaitility/4.0.3/awaitility-4.0.3.jar:/home/iid/.m2/repository/org/hamcrest/hamcrest/2.1/hamcrest-2.1.jar:/home/iid/.m2/repository/org/glassfish/jersey/ext/jersey-entity-filtering/2.26/jersey-entity-filtering-2.26.jar:/home/iid/.m2/repository/org/glassfish/jersey/media/jersey-media-json-jackson/2.26/jersey-media-json-jackson-2.26.jar:/home/iid/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.9.8/jackson-databind-2.9.8.jar:/home/iid/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.9.8/jackson-core-2.9.8.jar:/home/iid/.m2/repository/com/fasterxml/jackson/module/jackson-module-jaxb-annotations/2.9.8/jackson-module-jaxb-annotations-2.9.8.jar:/home/iid/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.9.8/jackson-annotations-2.9.8.jar:/home/iid/.m2/repository/javax/ws/rs/javax.ws.rs-api/2.1/javax.ws.rs-api-2.1.jar:/home/iid/.m2/repository/org/glassfish/jersey/inject/jersey-hk2/2.26/jersey-hk2-2.26.jar:/home/iid/.m2/repository/org/glassfish/hk2/hk2-locator/2.5.0-b36/hk2-locator-2.5.0-b36.jar:/home/iid/.m2/repository/org/glassfish/hk2/external/aopalliance-repackaged/2.5.0-b36/aopalliance-repackaged-2.5.0-b36.jar:/home/iid/.m2/repository/org/glassfish/hk2/hk2-api/2.5.0-b36/hk2-api-2.5.0-b36.jar:/home/iid/.m2/repository/javax/inject/javax.inject/1/javax.inject-1.jar:/home/iid/.m2/repository/org/glassfish/hk2/hk2-utils/2.5.0-b36/hk2-utils-2.5.0-b36.jar:/home/iid/.m2/repository/org/codehaus/woodstox/stax2-api/3.1.1/stax2-api-3.1.1.jar:/home/iid/.m2/repository/javax/xml/stream/stax-api/1.0-2/stax-api-1.0-2.jar:/home/iid/.m2/repository/xerces/xercesImpl/2.12.2/xercesImpl-2.12.2.jar:/home/iid/.m2/repository/xml-apis/xml-apis/1.4.01/xml-apis-1.4.01.jar:/home/iid/.m2/repository/org/apache/commons/commons-collections4/4.1/commons-collections4-4.1.jar com.clear2pay.bph.gpp.goldfish.functionaltests.systemtests.sanctionfilter.Pacs008SanctionFilterBypassTest
Dec 27, 2023 12:56:59 PM org.openqa.selenium.manager.SeleniumManager getDriverPath
INFO: Applicable driver not found; attempting to install with Selenium Manager (Beta)
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
ERROR StatusLogger Log4j2 could not find a logging implementation. Please add log4j-core to the classpath. Using SimpleLogger to log to the console...
1703678220602	geckodriver	INFO	Listening on 127.0.0.1:18601
1703678220871	mozrunner::runner	INFO	Running command: MOZ_CRASHREPORTER="1" MOZ_CRASHREPORTER_NO_REPORT="1" MOZ_CRASHREPORTER_SHUTDOWN="1" MOZ_NO_REMOTE="1" "/sna ... gins" "http://127.0.0.1:6208/,http://localhost:6208/,http://[::1]:6208/" "-no-remote" "-profile" "/tmp/rust_mozprofileaWH7uR"
org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Response code 500. Message: Failed to start browser /snap/firefox/current/firefox.launcher: no such file or directory 
Host info: host: 'iid.clear2pay.com', ip: '127.0.0.1'
Build info: version: '4.9.1', revision: 'eb2032df7f'
System info: os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.0-1160.102.1.el7.x86_64', java.version: '1.8.0_261'
Driver info: org.openqa.selenium.firefox.FirefoxDriver
Command: [null, newSession {capabilities=[Capabilities {acceptInsecureCerts: true, browserName: firefox, moz:debuggerAddress: true, moz:firefoxOptions: {}}]}]
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:136)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:94)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:68)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:165)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:183)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:158)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:543)
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:229)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:157)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:150)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:146)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:131)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:127)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:112)
	at com.clear2pay.bph.gpp.goldfish.functionaltests.systemtests.sanctionfilter.Pacs008SanctionFilterBypassTest.main(Pacs008SanctionFilterBypassTest.java:114)
```

### Operating System

CentOS

### Selenium version

Java 4.9.1

### What are the browser(s) and version(s) where you see this issue?

Firefox 102

### What are the browser driver(s) and version(s) where you see this issue?

GeckoDriver (don't know how to check the version)

### Are you using Selenium Grid?

I don't think so

**Comment by titusfortner** [comment]
Selenium has done a lot of work on automated driver management over the past year. Please update to the latest version and retry.
If it isn't working, please turn on logging and provide results: https://www.selenium.dev/documentation/webdriver/troubleshooting/logging/

**Comment by noelo-cohelo** [comment]
It seems to use the newest version of Selenium, I had to upgrade JDK - it wouldn't work on JDK 8.
It worked fine with JDK 21 and Selenium 4.16.1. What is interesting, it also worked with JDK 21 and Selenium 4.9.1.
Unfortunately I cannot upgrade JDK in our project since we still have some Java 6 code which is not compatible.

I just found this, probably related: https://github.com/mozilla/geckodriver/issues/2062.

There is even nicer workaround there, by creating a symbolic link between the place where GeckoDriver expects Firefox to be and the one it actually is.

The problem only exists when running the provided code from snap-installed IntelliJ Idea. When I tried from manually installed instance, it works fine.
